### PR TITLE
Add a test case for CopyTexImage2D behavior when source/target images

### PR DESCRIPTION
**Pull request by zhenyao**
are from the same texture.

This is a regression test for crbug.com/797235

### CODE REVIEW

**Comment by zhenyao** [comment]
@kenrussell @kainino0x @jdarpinian @vonture one of you please review

**Comment by kenrussell** [comment]
This test doesn't actually verify the contents of the destination texture level. Could you please add that checking? Otherwise the test is not that strict.

File: sdk/tests/conformance2/textures/misc/copy-texture-image-same-texture.html [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    wtu.fillTexture(gl, texture, 64, 64, [255, 0, 0, 255], 0, gl.RGBA, gl.UNSIGNED_BYTE, gl.RGBA8);
+    gl.generateMipmap(gl.TEXTURE_2D);
+    gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture, 1);
+    checkFramebuffer([gl.FRAMEBUFFER_COMPLETE]);
+
+    gl.copyTexImage2D(gl.TEXTURE_2D, 2, gl.LUMINANCE_ALPHA, 0, 0, 16, 16, 0);
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "copyTexImage2D should succeed.");
+    checkFramebuffer([gl.FRAMEBUFFER_INCOMPLETE_ATTACHMENT]);
+
+    checkTextureLevelColor(texture, 2, 16, 16, 0, 0, 16, 16, [255, 255, 255, 255]);
```
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/copy-texture-image-same-texture.html]: On Linux and Mac, this fails with [255, 255, 0, 255]. On Windows or Linux with --use-gl=angle, this passes.

I think the test is correct, but we have a bug in Chrome command buffer implementation.

I'll file a bug for this.

**Comment by zhenyao** [comment]
@kainino0x Ken is out of office. Can you review this?

File: sdk/tests/conformance2/textures/misc/copy-texture-image-same-texture.html [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    var fbo = gl.createFramebuffer();
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+
+    var texture = gl.createTexture();
+    wtu.fillTexture(gl, texture, 64, 64, [255, 0, 0, 255], 0, gl.RGBA, gl.UNSIGNED_BYTE, gl.RGBA8);
+    gl.generateMipmap(gl.TEXTURE_2D);
+    gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture, 1);
+    checkFramebuffer([gl.FRAMEBUFFER_COMPLETE]);
+
+    gl.copyTexImage2D(gl.TEXTURE_2D, 2, gl.RGBA8, 0, 0, 64, 64, 0);
```
Comment (kainino0x) [inline sdk/tests/conformance2/textures/misc/copy-texture-image-same-texture.html]: It took me a while to understand this, can you add a comment just to clarify?

    // Level 0 is 64x64. Reallocate level 2 at 64x64, so the framebuffer is no longer complete.

File: sdk/tests/conformance2/textures/misc/copy-texture-image-same-texture.html [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    var fbo = gl.createFramebuffer();
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+
+    var texture = gl.createTexture();
+    wtu.fillTexture(gl, texture, 64, 64, [255, 0, 0, 255], 0, gl.RGBA, gl.UNSIGNED_BYTE, gl.RGBA8);
+    gl.generateMipmap(gl.TEXTURE_2D);
+    gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture, 1);
+    checkFramebuffer([gl.FRAMEBUFFER_COMPLETE]);
+
+    gl.copyTexImage2D(gl.TEXTURE_2D, 2, gl.LUMINANCE_ALPHA, 0, 0, 16, 16, 0);
```
Comment (kainino0x) [inline sdk/tests/conformance2/textures/misc/copy-texture-image-same-texture.html]: // Level 0 is gl.RGBA. Reallocate level 2 with LUMINANCE_ALPHA, so the framebuffer is no longer complete.

**Comment by zhenyao** [comment]
Detailed comments added.

**Comment by kenrussell** [comment]
Great work. Thanks Mo for strengthening this test. It looks like it uncovers some longstanding bugs.

**Pull request merged**